aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/BaseHandlerStd.java

```diff
@@ -381,63 +385,95 @@ private Map<String, Parameter> getParametersToReset(
 
     private ProgressEvent<ResourceModel, CallbackContext> describeCurrentDBParameters(
             final ProgressEvent<ResourceModel, CallbackContext> progress,
-            final Collection<String> paramNames,
+            final List<String> paramNames,
             final Map<String, Parameter> currentParams,
             final AmazonWebServicesClientProxy proxy,
             final ProxyClient<RdsClient> proxyClient,
-            final String marker,
             final RequestLogger logger
     ) {
-        return proxy.initiate("rds::describe-db-parameters", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
-                .translateToServiceRequest((resourceModel) -> Translator.describeDbParametersRequest(resourceModel.getDBParameterGroupName(), paramNames, marker))
-                .makeServiceCall((request, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(request, proxyInvocation.client()::describeDBParameters))
-                .handleError((request, exception, client, model, ctx) ->
-                        Commons.handleException(
-                                ProgressEvent.progress(model, ctx),
-                                exception,
-                                DEFAULT_DB_PARAMETER_GROUP_ERROR_RULE_SET
-                        ))
-                .done((request, response, proxyInvocation, model, context) -> {
-                    currentParams.putAll(response.parameters().stream()
-                            .collect(Collectors.toMap(Parameter::parameterName, Function.identity())));
-                    if (StringUtils.isNullOrEmpty(response.marker())) {
-                        return ProgressEvent.progress(model, context);
+        for (final List<String> paramNamePartition : Lists.partition(paramNames, MAX_PARAMETER_FILTER_SIZE)) {
+            String marker = null;
+            try {
+                int page = 1;
+                do {
+                    if (page > MAX_PARAMETER_DESCRIBE_DEPTH) {
```

Comment:
This effectively limits RDS CFN at 2000 parameters, if default values are used. While I don't think we have any engine with this many parameters yet, I think it is possible that we have this many in the future. I.e. postgres13 has ~250 parameters while postgres14 more than doubled that with over 550.

IMO, potential bugs in AWS RDS will explode other tools like AWS CLI and it will trigger fast error correction from AWS RDS team, which makes this defensive code relatively limited in functionality.